lib/python/Screens/FlashManager.py

```diff
@@ -0,0 +1,600 @@
+							print("[FlashManager] OLD DEBUG: subfolder='%s'." % subfolder)
+							if isdir(subfolder) and not islink(subfolder) and not ismount(subfolder):
+								print("[FlashManager] OLD DEBUG: Next subfolder='%s'." % subfolder)
+								getImages(subfolder, [pathjoin(subfolder, x) for x in listdir(subfolder) if splitext(x)[1] == ".zip" and self.box in x])
+								for dir in [dir for dir in [pathjoin(subfolder, dir) for dir in listdir(subfolder)] if isdir(dir) and splitext(dir)[1] == ".unzipped"]:
+									try:
+										rmtree(dir)
+									except (IOError, OSError) as err:
+										print("[FlashManager] Error %d: Unable to remove directory '%s'!  (%s)" % (err.errno, dir, err.strerror))
+		imageList = []
+		for catagory in reversed(sorted(self.imagesList.keys())):
+			if catagory in self.expanded:
+				imageList.append(ChoiceEntryComponent("expanded", ((str(catagory)), "Expanded")))
+				for image in reversed(sorted(list(self.imagesList[catagory].keys()), key=lambda x: x.split(sep)[-1])):
+					imageList.append(ChoiceEntryComponent("verticalline", ((self.imagesList[catagory][image]["name"]), self.imagesList[catagory][image]["link"])))
+			else:
+				for image in self.imagesList[catagory].keys():
+					imageList.append(ChoiceEntryComponent("expandable", ((catagory), "Expanded")))
+					break
+		if imageList:
+			self["list"].setList(imageList)
+			if self.setIndex:
+				self["list"].moveToIndex(self.setIndex if self.setIndex < len(imageList) else len(imageList) - 1)
+				if self["list"].l.getCurrentSelection()[0][1] == "Expanded":
+					self.setIndex -= 1
+					if self.setIndex:
+						self["list"].moveToIndex(self.setIndex if self.setIndex < len(imageList) else len(imageList) - 1)
+				self.setIndex = 0
+			self.selectionChanged()
+		else:
+			self.session.openWithCallback(self.keyCancel, MessageBox, _("Error: Cannot find any images!"), type=MessageBox.TYPE_ERROR, timeout=3)
+
+	def keyCancel(self):
+		self.close()
+
+	def closeRecursive(self):
+		self.close(True)
+
+	def keyOk(self):
+		currentSelection = self["list"].l.getCurrentSelection()
+		if currentSelection[0][1] == "Expanded":
+			if currentSelection[0][0] in self.expanded:
+				self.expanded.remove(currentSelection[0][0])
+			else:
+				self.expanded.append(currentSelection[0][0])
+			self.getImagesList()
+		elif currentSelection[0][1] != "Loading":
+			self.session.openWithCallback(self.getImagesList, FlashImage, currentSelection[0][0], currentSelection[0][1])
+
+	def keyDistribution(self):
+		default = 0
+		distributionList = []
+		for index, distribution in enumerate(sorted(FEED_URLS.keys())):
+			distributionList.append((distribution, distribution))
+			if distribution == self.imageFeed:
+				default = index
+		message = _("Please select a distribution from which you would like to flash an image:")
+		self.session.openWithCallback(self.keyDistributionCallback, MessageBox, message, default=default, list=distributionList, windowTitle=_("Flash Manager"), simple=True)
```

Comment:
Why is the "simple" form being requested.  I thought we were going to remove this.